dds/idl/marshal_generator.cpp

```diff
@@ -2336,7 +2336,7 @@ bool marshal_generator::gen_struct(AST_Structure* node,
           "    return false;\n"
           "  }\n";
       }
-      if (repr.xcdr1 && may_be_parameter_list) {
+      if (repr.xcdr1) {
```

Comment:
Isn't this `if` inside of an `if (may_be_parameter_list)`, which means the extensibility here is mutable? I only removed `&& may_be_parameter_list` in this `if` because it's redundant.

Reply:
Yea I missed that. I put the wrong thing in there anyway.

Reply:
You can actually make it do it unconditionally since we need to eliminate use of the `repr` optimizations anyway.

Reply:
Yes, I'll remove the `if` statement.